liionpack/solvers.py

```diff
@@ -177,54 +184,68 @@ def solve(
 
         sim_start_time = ticker.time()
         lp.logger.notice("Starting step solve")
-        for step in tqdm(range(Nsteps), desc="Stepping simulation"):
-            # Solver specific steps
-            self.step_actors()
-            self.get_actor_output(step)
-
-            time += self.dt
-
-            # Calculate internal resistance and update netlist
-            temp_v = self.output[0, step, :]
-            temp_ocv = self.output[1, step, :]
-            # This could be used instead of Equivalent ECM resistance which has
-            # been changing definition
-            temp_Ri = (temp_ocv - temp_v) / self.shm_i_app[step, :]
-            # Make Ri more stable
-            current_cutoff = np.abs(self.shm_i_app[step, :]) < 1e-6
-            temp_Ri[current_cutoff] = 1e-12
-
-            netlist.loc[V_map, ("value")] = temp_ocv
-            netlist.loc[Ri_map, ("value")] = temp_Ri
-            netlist.loc[I_map, ("value")] = protocol[step]
-
-            if time <= end_time:
-                record_times.append(time)
-                V_node, I_batt = lp.solve_circuit_vectorized(netlist)
-                V_terminal.append(V_node[Terminal_Node][0])
-            if time < end_time:
-                I_app = I_batt[:] * -1
-                self.shm_i_app[step + 1, :] = I_app
-                self.inputs_dict = lp.build_inputs_dict(I_app, self.inputs)
-            # Stop if voltage limits are reached
-            if np.any(temp_v < v_cut_lower):
-                lp.logger.warning("Low voltage limit reached")
-                break
-            if np.any(temp_v > v_cut_higher):
-                lp.logger.warning("High voltage limit reached")
-                break
-
-            self.timestep += 1
+        self.fixed_Ri = np.ones(Nspm) * np.nan
+        with tqdm(total=Nsteps, desc="Stepping simulation") as pbar:
+            step = 0
+            while step < Nsteps:
+                # Solver specific steps
+                # self.resting = protocol[step] == 0.0
+                self.resting = (
+                    step > 0 and protocol[step] == 0.0 and protocol[step - 1] == 0.0
+                )
+                self.restarting = (
+                    step > 0 and protocol[step] != 0.0 and protocol[step - 1] == 0.0
+                )
+                self.step_actors()
+                self.get_actor_output(step)
+
+                # # Calculate internal resistance and update netlist
+                temp_v = self.output[0, step, :]
+                temp_ocv = self.output[1, step, :]
+                if not self.resting and not self.restarting:
```

Comment:
Could do with a comment explaining why you don't always update temp_Ri

Reply:
Added